examples/authorize_and_store_papi_smoldot.js

```diff
@@ -0,0 +1,136 @@
+    const bobApi = await ApiPromise.create({ provider: bobWs });
+    await bobApi.isReady;
+
+    // Create keyring and accounts
+    const keyring = new Keyring({ type: 'sr25519' });
+    const sudoAccount = keyring.addFromUri('//Alice');
+    const whoAccount = keyring.addFromUri('//Alice');
+
+    // Data
+    const who = whoAccount.address;
+    const transactions = 32;
+    const bytes = 64 * 1024 * 1024; // 64 MB
+
+    // Prepare data for storage
+    const dataToStore = "Hello, Bulletin with PAPI + Smoldot - " + new Date().toString();
+    const cid = cidFromBytes(dataToStore);
+
+    // Start smoldot with logging enabled
+    console.log('\n🚀 Starting smoldot...');
+    const client = smoldot.start({
+        maxLogLevel: 0, // 0=off, 1=error, 2=warn, 3=info, 4=debug, 5=trace
+        logCallback: (level, target, message) => {
+            const levelNames = ['ERROR', 'WARN', 'INFO', 'DEBUG', 'TRACE'];
+            const levelName = levelNames[level - 1] || 'UNKNOWN';
+            console.log(`[smoldot:${levelName}] ${target}: ${message}`);
+        }
+    });
+
+    // Note: In real usage, this step is not required — the chain spec with bootNodes should be included as part of the dApp.
+    //       For local testing, we use this to fetch the actual chain spec from the local node.
+    // Get chain spec from Bob node and remove protocolId to allow smoldot to sync with local chain.
+    const chainSpec = (await bobApi.rpc.syncstate.genSyncSpec(true)).toString();
+    const chainSpecObj = JSON.parse(chainSpec);
+    chainSpecObj.protocolId = null;
+    const modifiedChainSpec = JSON.stringify(chainSpecObj);
+
+    await client
+        .addChain({ chainSpec: modifiedChainSpec })
+        .then(async (chain) => {
+            // Give smoldot a moment to sync
+            console.log("⏳ Waiting for smoldot to sync...");
+            await new Promise(resolve => setTimeout(resolve, 12000));
+            
+            const authorizeTx = aliceApi.tx.transactionStorage.authorizeAccount(
+                who,
+                transactions,
+                bytes
+            );
+            const sudoTx = aliceApi.tx.sudo.sudo(authorizeTx);
+            const signedAuthTx = await sudoTx.signAsync(sudoAccount);
+            
+            const authBlockHash = await submitAndWatch(chain, signedAuthTx, 'authorizeAccount');
+            console.log('✅ Authorized in block:', authBlockHash);
+            
+            return chain;
+        })
+        .then(async (chain) => {
+            const dataBytes = Buffer.from(dataToStore);
+            const storeTx = aliceApi.tx.transactionStorage.store(dataBytes);
+            const signedStoreTx = await storeTx.signAsync(whoAccount);
+            
+            const storeBlockHash = await submitAndWatch(chain, signedStoreTx, 'store');
+            console.log('✅ Stored data with CID:', cid);
+            console.log('   In block:', storeBlockHash);
+            
```

Comment:
@x3c41a please copy `fetchCid` from here: https://github.com/paritytech/polkadot-bulletin-chain/pull/100/files#diff-e3a5dd74ebd78c0734ad7bac27892935628227b0da5e1375038908c9e717ffadR173-R179

```suggestion
            
            // Download the content from IPFS HTTP gateway.
            const cid = cidFromBytes(dataBytes);
            const downloaded = await fetchCid(cid);
            assert.deepStrictEqual(dataBytes.length, downloaded.length)
            
```